src/main/java/io/stargate/sgv2/jsonapi/service/operation/tables/WhereCQLClauseAnalyzer.java

```diff
@@ -237,6 +239,35 @@ private void checkFilteringOnComplexColumns(Map<CqlIdentifier, TableFilter> iden
     }
   }
 
+  /**
+   * Check if any $in filter is used.
+   *
+   * <p>For UpdateOne and DeleteOne table commands, the use of $in/$nin can affect multiple rows, so
+   * add a check rule to ban the usage.
+   */
+  private void checkNoInFilterUsage(Map<CqlIdentifier, TableFilter> identifierToFilter) {
+
+    var inFilterColumns =
+        identifierToFilter.entrySet().stream()
+            .filter(
+                entry -> {
+                  TableFilter tableFilter = entry.getValue();
```

Comment:
nit: could this be one line ? we are not using the tableFilter var

Reply:
fixed